File: Lab_2/app/src/main/java/com/example/bmi_app/Bmi.java
package com.example.bmi_app;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class Bmi extends AppCompatActivity {
    private EditText weight;
    private EditText height;
    private TextView bmi;
    private TextView result;
    private Button calculate;
    private String description = "";
    private String BMI;
    static String bmiValueString;

    public String getBMI() {
        return BMI;
    }

    public void setBMI(String BMI) {
        this.BMI = BMI;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.bmi);

        Button backHome = findViewById(R.id.home_button);
        Button goToKcal = findViewById(R.id.kcal_button);
        Button goToMenu = findViewById(R.id.menu_button);
        backHome.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), Home.class);
                startActivity(intent);
            }
        });
        goToKcal.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), Kcal.class);
                startActivity(intent);
            }
        });
        goToMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), Menu.class);
                if(BMI==null || BMI.isEmpty()){
                    result.setText(R.string.bmiNull);
                }else
                    startActivity(intent);
            }
        });

        weight = findViewById(R.id.weight);
        height = findViewById(R.id.height);
        bmi = findViewById(R.id.bmi);
        result = findViewById(R.id.result);
        calculate = findViewById(R.id.calculate);

        calculate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculateBMI();

            }
        });
    }

    public String calculateBMI(){
        String inputWeight = weight.getText().toString();
        String inputHeight = height.getText().toString();

        if(inputWeight.isEmpty() || inputHeight.isEmpty() || inputWeight.equals("0") || inputHeight.equals("0")){
            description = "Uzupełnij poprawnie dane";
            result.setText(description);
        }else {

            double weightValue = Double.parseDouble(inputWeight);
            double heightValue = Double.parseDouble(inputHeight) / 100;

            double bmiValue = (double) (Math.round((weightValue / (Math.pow(heightValue, 2))) * 100)) / 100;


            bmiValueString = Double.toString(bmiValue);

            bmi.setText(bmiValueString);

            if (bmiValue < 16) {
                bmiValueString = "Wygłodzenie";
            } else if (bmiValue >= 16 && bmiValue < 17) {
                bmiValueString = "Wychudzenie";
            } else if (bmiValue >= 17 && bmiValue < 18.5) {
                bmiValueString = "Niedowaga";
            } else if (bmiValue >= 18.5 && bmiValue < 25) {
                bmiValueString = "Waga prawidłowa";
            } else if (bmiValue >= 25 && bmiValue < 30) {
                bmiValueString = "Nadwaga";
            } else if (bmiValue >= 30 && bmiValue < 35) {
                bmiValueString = "Otyłość I stopnia";
            } else if (bmiValue >= 35 && bmiValue < 40) {
                bmiValueString = "Otyłość II stopnia";
            } else {
                bmiValueString = "Otyłość III stopnia";
            }
            result.setText(bmiValueString);
            BMI = bmiValueString;
            setBMI(BMI);

        }
        return bmiValueString;
    }

}

File: Lab_4_Lab_5/app/src/main/java/com/example/bmi_app/Bmi.kt
package com.example.bmi_app

import androidx.appcompat.app.AppCompatActivity

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle

import kotlinx.android.synthetic.main.bmi.*
import java.lang.String.format
import kotlin.math.pow

class Bmi : AppCompatActivity() {
    companion object {
        var bmiValueString = ""
    }
    var BMI = ""

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.bmi)

        calculateBMI()

        home_button.setOnClickListener {
            val intent = Intent (this, MainActivity::class.java)
            startActivity(intent)
        }

        kcal_button.setOnClickListener {
            val intent = Intent (this, com.example.bmi_app.Kcal::class.java)
            startActivity(intent)
        }

        menu_button.setOnClickListener {
            val intent = Intent(this, Menu::class.java)
            if(BMI.isEmpty()){
                result.text = "Najpierw Oblicz BMI"
            }else
            startActivity(intent)
        }
    }

    @SuppressLint("DefaultLocale", "SetTextI18n")
    private fun calculateBMI():String{
//        val inputWeight = weight.text.toString()
//        val inputHeight = height.text.toString()

        calculate.setOnClickListener {
            val inputWeight = weight.text.toString()
            val inputHeight = height.text.toString()
            if(inputWeight.isEmpty() || inputHeight.isEmpty() || inputHeight.equals("0") || inputWeight.equals("0")){
                result.text = "Uzupełnij poprawnie dane"
            }else{
                val weightValue = inputWeight.toDouble()
                val heightValue = inputHeight.toDouble()
                val bmiValue = format("%.1f",((weightValue/((heightValue.pow(2))/100))*100)).toDouble()

                bmiValueString = bmiValue.toString()

                bmi.setText(bmiValueString)

                bmiValueString = when (bmiValue){
                    in 0.0..15.9 -> "Wygłodzenie"
                    in 16.0..16.9 -> "Wychudzenie"
                    in 17.0..18.4 -> "Niedowaga"
                    in 18.5..24.9 -> "Waga prawidłowa"
                    in 25.0..29.9 -> "Nadwaga"
                    in 30.0..34.9 -> "Otyłość I stopnia"
                    in 35.0..40.0 -> "Otyłość II stopnia"
                    in 40.0..100.0 -> "Otyłość III stopnia"
                    else -> "Błędne dane"
                }
                result.text = bmiValueString
                BMI = bmiValueString

            }
        }
        return bmiValueString
    }
}
